assets/javascripts/discourse/components/chat-message.js

```diff
@@ -764,6 +792,35 @@ export default Component.extend({
     ).catch(popupAjaxError);
   },
 
+  @action
+  toggleBookmark() {
+    return openBookmarkModal(
+      this.bookmark ||
+        Bookmark.create({
+          bookmarkable_type: "ChatMessage",
+          bookmarkable_id: this.message.id,
+          user_id: this.currentUser.id,
+        }),
+      {
+        onAfterSave: (savedData) => {
+          const bookmark = Bookmark.create(savedData);
+          this.set("message.bookmark", savedData);
```

Comment:
it's possible you currently can't do differently due to chat design, but this bookmark and message.bookmark at the same time seems like a smell to me.

Reply:
I just did it this way because I didn't want to override `message.bookmark` which comes down from the serializer, but thinking about it more I can probably just do that without problems :+1: